src/Text/Parsing/StringParser/String.purs

```diff
@@ -46,3 +56,35 @@ string :: String -> Parser String
 string nt = Parser (\s fc sc -> case s of
   { str = str, pos = i } | indexOf' nt i str == Just i -> sc nt { str: str, pos: i + length nt }
   { pos = i } -> fc i (ParseError $ "Expected '" ++ nt ++ "'."))
+
+-- | Match a character satisfying the given predicate.
+satisfy :: (Char -> Boolean) -> Parser Char
+satisfy f = try do
+  c <- anyChar
+  if f c
+     then return c
+     else fail "Character did not satisfy predicate"
+
+-- | Match the specified character.
+char :: Char -> Parser Char
+char c = satisfy (== c)
+
+-- | Match many whitespace characters.
+whiteSpace :: Parser String
+whiteSpace = do
+  cs <- many $ satisfy \ c -> c == 'n' || c == 'r' || c == ' ' || c == '\t'
+  return $ foldMap toString cs
```

Comment:
Yes, I think your `satisfy` error is fine, but maybe you could use `<?>` to provide better errors here.